### PR TITLE
Move methods again 🤘😃

### PR DESCRIPTION
**Pull request by julienbourdeau**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | no
| New feature?      | no    
| BC breaks?        | no     
| Related Issue     | Fix #...  
| Need Doc update   | yes/no

## Describe your change

- [x] Introduce MultiResponse which is just a wrapper around more responses
- [x] Refactor copy*To, moveTo to be part of the client
- [x] Introduce `AccountClient::copyIndex($index1, $index2)`

### CODE REVIEW

File: src/AccountClient.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Algolia\AlgoliaSearch;
+
+use Algolia\AlgoliaSearch\Response\MultiResponse;
+
+class AccountClient
+{
+    public static function copyIndex(SearchIndex $srcIndex, SearchIndex $destIndex)
```
Comment (nunomaduro) [inline src/AccountClient.php]: What happens if the destination index already exists?

File: src/AccountClient.php [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Algolia\AlgoliaSearch;
+
+use Algolia\AlgoliaSearch\Response\MultiResponse;
+
+class AccountClient
```
Comment (nunomaduro) [inline src/AccountClient.php]: Shouldn't this be final?

File: src/AccountClient.php (hunk 1 above)
Comment (nunomaduro) [inline src/AccountClient.php]: There is a reason for making this `static` ?

File: src/Response/MultiResponse.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Response;
+
+class MultiResponse extends AbstractResponse implements \Iterator, \Countable
```
Comment (nunomaduro) [inline src/Response/MultiResponse.php]: Nice one! Can you add this response on methods that are returning more then one response? Example: `replaceAllObjects`?

File: src/AccountClient.php (hunk 2 above)
Comment (julienbourdeau) [reply src/AccountClient.php]: it should

File: src/AccountClient.php (hunk 1 above)
Comment (julienbourdeau) [reply src/AccountClient.php]: Currenly, Acccount doesnt have any state and doesnt have any dependencies. So I think static for now is good

File: src/Response/MultiResponse.php (hunk 3 above)
Comment (julienbourdeau) [reply src/Response/MultiResponse.php]: Normally I did it everywhere
(replaceAllObjects: https://github.com/algolia/algoliasearch-client-php/pull/481/files#diff-8e5cca1a0496ef5410c39a80df3a03e8R230)

File: src/AccountClient.php [hunk 4]
```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Algolia\AlgoliaSearch;
+
+use Algolia\AlgoliaSearch\Exceptions\NotFoundException;
+use Algolia\AlgoliaSearch\Response\MultiResponse;
+
+final class AccountClient
+{
+    public static function copyIndex(SearchIndex $srcIndex, SearchIndex $destIndex, $requestOptions = array())
```
Comment (Ant-hem) [inline src/AccountClient.php]: Correct me if I am wrong but `$requestOptions `are never used here?

Comment (julienbourdeau) [reply src/AccountClient.php]: Yes, never used yet but required for future-proof method signature. All methods, unless there is really a strong reason not to, have a RequestOptions last argument

File: src/AccountClient.php (hunk 1 above)
Comment (julienbourdeau) [reply src/AccountClient.php]: Great catch! as discussed, the method will check if the destination exists and fail if it does.

File: src/AccountClient.php [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
```
Comment (nunomaduro) [inline src/AccountClient.php]: You need to check if the indexes are from different accounts.

Comment (julienbourdeau) [reply src/AccountClient.php]: I don't like to expose the appid from the Index but I see why we need to do that 👌

**Pull request merged**